### ISSUE TITLE
Amadron tablet is not working on servers in 1.20.4 Neoforge

### ISSUE THREAD

**Issue by Saereth** [TeamPneumatic/pnc-repressurized#1313]
### Describe the bug

Amadron tablet isn't working. The trades show, you can link to a chest but the order button stays grayed out.
<img width="615" alt="image" src="https://github.com/TeamPneumatic/pnc-repressurized/assets/9200909/34439404-ea7b-4274-986a-421bc0b88667">

### How to reproduce the bug

Load up pnc on a neo server, try to place a valid order after binding to a chest and placing emeralds in the chest.

### Expected behavior

It should work, it does work on single player.

### Additional details

Also tried with your test build: pneumaticcraft-repressurized-7.0.2+mc1.20.4-20240615.084350-1

Here is the log:
https://mclo.gs/kUFaOPA

the error of note, this spams the client's local.log anytime they adjust the order in basket
```
18Jun2024 17:52:57.788] [Render thread/ERROR] [net.minecraft.util.thread.BlockableEventLoop/FATAL]: Error executing task on Client
java.util.concurrent.CompletionException: java.lang.IndexOutOfBoundsException: Index 40 out of bounds for length 33
	at java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:315) ~[?:?]
	at java.util.concurrent.CompletableFuture.completeThrowable(CompletableFuture.java:320) ~[?:?]
	at java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1770) ~[?:?]
	at net.minecraft.util.thread.BlockableEventLoop.doRunTask(BlockableEventLoop.java:146) ~[client-1.20.4-20231207.154220-srg.jar%23705!/:?]
	at net.minecraft.util.thread.ReentrantBlockableEventLoop.doRunTask(ReentrantBlockableEventLoop.java:23) ~[client-1.20.4-20231207.154220-srg.jar%23705!/:?]
	at net.minecraft.util.thread.BlockableEventLoop.pollTask(BlockableEventLoop.java:120) ~[client-1.20.4-20231207.154220-srg.jar%23705!/:?]
	at net.minecraft.util.thread.BlockableEventLoop.runAllTasks(BlockableEventLoop.java:109) ~[client-1.20.4-20231207.154220-srg.jar%23705!/:?]
	at net.minecraft.client.Minecraft.runTick(Minecraft.java:1223) ~[client-1.20.4-20231207.154220-srg.jar%23705!/:?]
	at net.minecraft.client.Minecraft.run(Minecraft.java:818) ~[client-1.20.4-20231207.154220-srg.jar%23705!/:?]
	at net.minecraft.client.main.Main.main(Main.java:244) ~[neoforge-20.4.237.jar:?]
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:?]
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77) ~[?:?]
	at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:?]
	at java.lang.reflect.Method.invoke(Method.java:568) ~[?:?]
	at net.neoforged.fml.loading.targets.CommonLaunchHandler.runTarget(CommonLaunchHandler.java:136) ~[loader-2.0.17.jar:2.0]
	at net.neoforged.fml.loading.targets.CommonLaunchHandler.clientService(CommonLaunchHandler.java:124) ~[loader-2.0.17.jar:2.0]
	at net.neoforged.fml.loading.targets.CommonClientLaunchHandler.runService(CommonClientLaunchHandler.java:24) ~[loader-2.0.17.jar:2.0]
	at net.neoforged.fml.loading.targets.CommonLaunchHandler.lambda$launchService$4(CommonLaunchHandler.java:118) ~[loader-2.0.17.jar:2.0]
	at cpw.mods.modlauncher.LaunchServiceHandlerDecorator.launch(LaunchServiceHandlerDecorator.java:30) ~[modlauncher-10.0.9.jar:?]
	at cpw.mods.modlauncher.LaunchServiceHandler.launch(LaunchServiceHandler.java:53) ~[modlauncher-10.0.9.jar:?]
	at cpw.mods.modlauncher.LaunchServiceHandler.launch(LaunchServiceHandler.java:71) ~[modlauncher-10.0.9.jar:?]
	at cpw.mods.modlauncher.Launcher.run(Launcher.java:108) ~[modlauncher-10.0.9.jar:?]
	at cpw.mods.modlauncher.Launcher.main(Launcher.java:78) ~[modlauncher-10.0.9.jar:?]
	at cpw.mods.modlauncher.BootstrapLaunchConsumer.accept(BootstrapLaunchConsumer.java:26) ~[modlauncher-10.0.9.jar:?]
	at cpw.mods.modlauncher.BootstrapLaunchConsumer.accept(BootstrapLaunchConsumer.java:23) ~[modlauncher-10.0.9.jar:?]
	at cpw.mods.bootstraplauncher.BootstrapLauncher.main(BootstrapLauncher.java:141) ~[bootstraplauncher-1.1.2.jar:?]
Caused by: java.lang.IndexOutOfBoundsException: Index 40 out of bounds for length 33
	at jdk.internal.util.Preconditions.outOfBounds(Preconditions.java:64) ~[?:?]
	at jdk.internal.util.Preconditions.outOfBoundsCheckIndex(Preconditions.java:70) ~[?:?]
	at jdk.internal.util.Preconditions.checkIndex(Preconditions.java:266) ~[?:?]
	at java.util.Objects.checkIndex(Objects.java:361) ~[?:?]
	at java.util.ArrayList.get(ArrayList.java:427) ~[?:?]
	at me.desht.pneumaticcraft.common.inventory.AbstractPneumaticCraftMenu.updateField(AbstractPneumaticCraftMenu.java:100) ~[pneumaticcraft-repressurized-7.0.2+mc1.20.4-20240615.084350-1.jar%23626!/:?]
	at me.desht.pneumaticcraft.client.util.ClientUtils.syncViaOpenContainerScreen(ClientUtils.java:340) ~[pneumaticcraft-repressurized-7.0.2+mc1.20.4-20240615.084350-1.jar%23626!/:?]
	at me.desht.pneumaticcraft.common.network.PacketUpdateGui.lambda$handle$0(PacketUpdateGui.java:62) ~[pneumaticcraft-repressurized-7.0.2+mc1.20.4-20240615.084350-1.jar%23626!/:?]
	at net.minecraft.util.thread.BlockableEventLoop.lambda$submitAsync$0(BlockableEventLoop.java:59) ~[client-1.20.4-20231207.154220-srg.jar%23705!/:?]
	at java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1768) ~[?:?]
	... 23 more
```

### Which Minecraft version are you using?

1.20

### Which version of PneumaticCraft: Repressurized are you using?

pneumaticcraft-repressurized-7.0.0

**Comment by desht** [comment]
This works fine for me, both in dev and on a dedicated test server.

I also don't get the log spam when adjusting orders...

The client error is likely the cause of not being able to place the order, but I'm not sure what the conditions are to trigger that...

**Comment by desht** [comment]
Sync issue with some custom trades, fixed in 7.0.1
